conda_smithy/templates/run_docker_build.tmpl

```diff
@@ -52,5 +53,18 @@ upload_or_check_non_existence /recipe_root {{ owner }} --channel={{ channel }} |
 {%- endfor -%}
 {%- endblock -%}
 
+# signal build-done with a file to protect against
+# an early exit with status=0 appearing to have succeeded
+touch /feedstock_root/build_artefacts/conda-forge-build-done
 EOF
 
+docker run -i \
```

Comment:
Given we are giving up redirection, what do you think about enabling TTY? Have already had a few issues where not having a TTY has caused problems (e.g. `ncurses`).